src/components/Chat/Chat.tsx

```diff
@@ -902,7 +902,7 @@ export class Chat extends React.Component<ChatProperties, ChatState> {
 
 export class EmbeddedChat extends React.PureComponent<ChatProperties, {}> {
     render() {
-        return <Card className="Card EmbeddedChat"><Chat {...this.props} /></Card>;
+        return <Card className="Card EmbeddedChat"><Chat key={this.props.channel} {...this.props} /></Card>;
```

Comment:
The `Chat` component has more complicated side-effects than the table. I figured if we just reload the entire component when the channel changes, that was the approach with the least risk of breaking things. I'm happy to take a stab at fixing it properly in the `Chat` component itself if you want, though.